pkg/utils/tekton/controller.go

```diff
@@ -87,6 +92,63 @@ func (k KubeController) RunVerifyTask(taskName, image string, taskTimeout int) (
 	return k.createAndWait(tr, taskTimeout)
 }
 
+func (k KubeController) AwaitAttestationAndSignature(image string, timeout time.Duration) error {
+	imageInfo := strings.Split(image, "/")
+	namespace := imageInfo[1]
+	imageName := imageInfo[2]
+	latestImageName := imageName + ":latest"
+
+	tags := &unstructured.UnstructuredList{}
+	tags.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "image.openshift.io",
+		Kind:    "ImageStreamTag",
+		Version: "v1",
+	})
+
+	return wait.PollImmediate(time.Second, timeout, func() (done bool, err error) {
+		err = k.Commonctrl.KubeRest().List(context.TODO(), tags, &k8sclient.ListOptions{Namespace: namespace})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return false, nil
+			} else {
+				return true, err
+			}
+		}
+
+		// search for the resulting task image's hash from the name in the metadata
+		var hash string
+		for _, tag := range tags.Items {
+			if tag.GetName() == latestImageName {
+				if name, found, err := unstructured.NestedString(tag.Object, "image", "metadata", "name"); err == nil && found {
+					hash = imageName + ":" + strings.Replace(name, ":", "-", 1)
+					break
+				}
+			}
+		}
+
+		// we didn't find the image from the task, should we err instead?
+		if hash == "" {
+			return false, nil
+		}
+
+		// loop again to see if .att and .sig tags are present
+		found := 0
```

Comment:
Should this loop be a separate function? This could be helpful for more chains testing.